### PR TITLE
Newsletters: add 156 (2021-07-07)

### PR DESCRIPTION
**Pull request by harding**
- [x] Add releases/RCs if any @harding
- [x] Add topic links @harding 
- [x] `Bitcoin Core #19651` wallet: importdescriptors update existing @glozow 
- [x] `C-Lightning #4610` opts: Add option to register extra TLV types to accept @dongcarl 
- [x] `Eclair #1854` Decode warning messages @bitschmidty 
- [x] `BIPs #1137` BIP 86: Key Derivation for Single Key P2TR Outputs @jnewbery 

Note: now that we have three articles in the taproot series, I've added a blog post containing all of them, similar to what we did for the bech32 series.  I've backdated the post to the start of the taproot series (6/23).

### CODE REVIEW

File: _posts/en/newsletters/2021-07-07-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    Descriptors were originally implemented for Bitcoin Core and have
+    seen increased adoption over the past year by other projects.
+    They're poised to see a significant increase in use as wallets begin
+    to explore the flexibility enabled by [taproot][topic taproot] and
+    the ability to simplify access to the flexible scripts through tools
+    like [miniscript][topic miniscript].
+
+<!-- FIXME:harding Gentry uses a lowercase leading character (bLIPs).  I
+asked in IRC why, but unless there's a *really* compelling reason, I'd
```
Comment (jnewbery) [inline _posts/en/newsletters/2021-07-07-newsletter.md:39]: ACK referring to these as BLIPs.

File: _posts/en/newsletters/2021-07-07-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    ![Zero-conf channel illustration](/img/posts/2021-07-zeroconf-channels.png)
+
+    Some LN implementations already support the idea in a
+    non-standardized way and all discussion participants seemed to favor
+    standardizing it.  The exact details to use were still being
+    discussed at the time of writing.
+
+## Preparing for taproot #3: taproot descriptors
+
+*A weekly [series][series preparing for taproot] about how developers
```
Comment (jnewbery) [inline _posts/en/newsletters/2021-07-07-newsletter.md:92]: We should also update the introductory text here: https://bitcoinops.org/en/newsletters/2021/06/23/#preparing-for-taproot-1-bech32m-sending-support and here: https://bitcoinops.org/en/newsletters/2021/06/30/#preparing-for-taproot-2-is-taproot-even-worth-it-for-single-sig to point to the series blog post.

File: _includes/specials/taproot/en/02-descriptors.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+following descriptor is sufficient:
+
+    tr(<key>)
+
+The same syntax may also be used for multisignatures and [threshold
+signatures][topic threshold signature].  For example, Alice, Bob, and
+Carol aggregate their keys using [MuSig][topic musig] and then pay to
+`tr(<combined_key>)`.
+
+Unintuitively, the key specified by `tr(<key>)` won't be the key encoded
```
Comment (jnewbery) [inline _includes/specials/taproot/en/02-descriptors.md:31]: Perhaps:

```suggestion
Unintuitively, the `key` specified by `tr(<key>)` won't be the key encoded
```

File: _includes/specials/taproot/en/02-descriptors.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+in the address.  The `tr()` descriptor follows a safety recommendation
+from [BIP341][] to use an internal key that commits to an unspendable
```
Comment (jnewbery) [inline _includes/specials/taproot/en/02-descriptors.md:33]: Maybe link directly to https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#cite_note-22

File: _includes/specials/taproot/en/02-descriptors.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+A tree can be specified as an optional second parameter to the
+descriptor template we used before.  For example if Alice wants to be
+able to spend via keypath, but she also wants to allow Bob, Carol, Dan,
+and Edmond spend via a scriptpath that generates an audit trail for
+her (but not for third-party chain surveilance), Alice can use the
+following descriptor:
+
+    tr( <a_key> , { {pk(<b_key>),pk(<c_key>)} , {pk(<d_key>),pk(<e_key>)} )
+
+The above features are everything that's absolutely required to use
```
Comment (jnewbery) [inline _includes/specials/taproot/en/02-descriptors.md:57]: Maybe remove "absolutely", or maybe "The above features are all that's required ..."

File: _includes/specials/taproot/en/02-descriptors.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+- **Timelocks, hashlocks, and pointlocks:** these powerful constructions
+  used in LN, [DLCs][topic dlc], [coinswaps][topic coinswap], and many
+  other protocols can currently only be described via the `raw()`
+  function.  Adding support for them directly to descriptors is
+  possible, but it may be the case that support will be added instead
+  through descriptor's sibling project, [miniscript][topic miniscript].
+  Integration of miniscript into Bitcoin Core is still an ongoing
+  project, but we expect its innovations to spread to other wallets in
+  the same way that tools like PSBTs and miniscript already have.
```
Comment (jnewbery) [inline _includes/specials/taproot/en/02-descriptors.md:92]: Should this be:

```suggestion
  the same way that tools like PSBTs and descriptors already have.
```

?

File: _posts/en/newsletters/2021-07-07-newsletter.md [hunk 7]
```diff
@@ -118,7 +118,12 @@ BOLTs][bolts repo].*
   extend descriptor ranges, reactivate inactive descriptors, and make other
   updates using the `importdescriptors` wallet RPC.
 
-- [C-Lightning #4610][] opts: Add option to register extra TLV types to accept FIXME:dongcarl
+- [C-Lightning #4610][] adds an `--experimental-accept-extra-tlv-types`
+  command-line option which allows users to specify a list of even TLV types
+  that `lightningd` should pass through for plugins to handle. Previously,
+  `lightningd` considered all evenly-typed TLV messages to be invalid. This
+  change allows for plugins to define and handle their own custom TLV types
+  unknown to `lightningd`.
```
Comment (dongcarl) [inline _posts/en/newsletters/2021-07-07-newsletter.md:126]: Not 100% sure if the rationale described in this last sentence is accurate. Can anyone sanity-check?

Comment (harding) [reply _posts/en/newsletters/2021-07-07-newsletter.md]: LGTM.  I added two words elsewhere in the description for clarity.

**Comment by harding** [comment]
@bitschmidty @jnewbery @glozow @dongcarl PR writeups look great, thanks!  @glozow thanks for also adding the link in the descriptors topic; that's not necessary if you don't want to do it in the future---I usually batch those at the end---but it's appreciated if you'd like to continue doing so.

@jnewbery thanks for the review!  I think all comments have been addressed.

**Comment by jnewbery** [comment]
Looks great. All review comments have been addressed and @glozow's and @dongcarl's write-ups look good to me. Thanks all!

File: _includes/specials/taproot/en/02-descriptors.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+a basic checksum, making them convenient for backing up address
+information, copying it between different wallets, or sharing it between
+wallets that are cooperating to provide multiple signatures.
+
+Although descriptors are currently only used by a few projects, they and
+the related [miniscript][topic miniscript] project have the potential to
+significantly improve interoperability between different wallets and
+tools.  This will become increasingly important as more users take
+advantage of the benefits of taproot to improve their security through
+[multisignatures][topic multisignature] and their resilency through
```
Comment (adamjonas) [inline _includes/specials/taproot/en/02-descriptors.md:14]:
```suggestion
[multisignatures][topic multisignature] and their resiliency through
```

File: _includes/specials/taproot/en/02-descriptors.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        / \
+       /   \
+      / \ / \
+      B C D E
+
+A tree can be specified as an optional second parameter to the
+descriptor template we used before.  For example if Alice wants to be
+able to spend via keypath, but she also wants to allow Bob, Carol, Dan,
+and Edmond spend via a scriptpath that generates an audit trail for
+her (but not for third-party chain surveilance), Alice can use the
```
Comment (adamjonas) [inline _includes/specials/taproot/en/02-descriptors.md:52]:
```suggestion
her (but not for third-party chain surveillance), Alice can use the
```

File: _includes/specials/taproot/en/02-descriptors.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+completely describing expected policies:
+
+- **Voided keypaths:** some users may want to prevent usage of keypath
+  spending in order to force scriptpath spending.  That can be done now
+  by using an unspendable key as the first parameter to `tr()`, but it'd
+  be nice to allow wallets to store this preference in the descriptor
+  itself and have it compute a privacy-preserving unspendable keypath.
+
+- **Tapscript multisig:** for legacy and v0 segwit, the `multi()` and
+  `sortedmulti()` descriptors can be used to describe use of the
```
Comment (adamjonas) [inline _includes/specials/taproot/en/02-descriptors.md:69]: Read this through a few times. It's clunky, I think the `used` and `use` being used in close use.

**Comment by harding** [comment]
@adamjonas Thanks!  Edits made.

Also pushed the RC for LND and the topic links.

**Comment by jnewbery** [comment]
ACK. All looks good. Thanks @harding!

**Comment by bitschmidty** [comment]
Great newsletter, team! 🙏  @harding @dongcarl @jnewbery and 👋 @glozow for authoring. @adamjonas @dongcarl @jnewbery thanks for the 🕵️

**Pull request merged**